api/handler/v1/runtime.go

```diff
@@ -729,6 +729,50 @@ func (sv *RuntimeServiceServer) ListResourceSpecification(ctx context.Context, r
 	}, nil
 }
 
+func (sv *RuntimeServiceServer) Replay(ctx context.Context, req *pb.ReplayRequest) (*pb.ReplayResponse, error) {
+	projectRepo := sv.projectRepoFactory.New()
+	projSpec, err := projectRepo.GetByName(req.GetProjectName())
+	if err != nil {
+		return nil, status.Error(codes.NotFound, fmt.Sprintf("%s: project %s not found", err.Error(), req.GetProjectName()))
+	}
+
+	namespaceRepo := sv.namespaceRepoFactory.New(projSpec)
+	namespaceSpec, err := namespaceRepo.GetByName(req.GetNamespace())
+	if err != nil {
+		return nil, status.Error(codes.NotFound, fmt.Sprintf("%s: namespace %s not found", err.Error(), req.GetNamespace()))
+	}
+
+	jobSpec, err := sv.jobSvc.GetByName(req.GetJobName(), namespaceSpec)
+	if err != nil {
+		return nil, status.Error(codes.NotFound, fmt.Sprintf("%s: failed to find the job %s for namespace %s", err.Error(),
+			req.GetJobName(), req.GetNamespace()))
+	}
+
+	startDate, err := time.Parse(job.DateFormat, req.StartDate)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse replay start date(e.g. %s): %v", job.DateFormat, err)
+	}
+
+	endDate := startDate
+	if req.EndDate != "" {
+		if endDate, err = time.Parse(job.DateFormat, req.EndDate); err != nil {
+			return nil, fmt.Errorf("unable to parse replay end date(e.g. %s): %v", job.DateFormat, err)
+		}
+	}
+	if endDate.Before(startDate) {
+		return nil, errors.New("replay end date cannot be before start date")
```

Comment:
use `status` package here instead of returning `error`

Reply:
fixed